### PR TITLE
AgPasture root organ refactor

### PR DESCRIPTION
**Pull request by hol353**
Working on #4970

### CODE REVIEW

File: Models/AgPasture/PastureBelowGroundOrgan.cs [hunk 1]
```diff
@@ -2,91 +2,74 @@
...
     using System;
     using System.Linq;
-    using Newtonsoft.Json;
 
     /// <summary>Describes a generic below ground organ of a pasture species.</summary>
     [Serializable]
     public class PastureBelowGroundOrgan : Model
     {
+        /// <summary>Minimum significant difference between two values.</summary>
+        private const double Epsilon = 0.000000001;
```
Comment (hol430) [inline Models/AgPasture/PastureBelowGroundOrgan.cs:17]: Lower case e here...also why is this required? Is this used by code which should be using `MathUtilities`?

File: Models/AgPasture/PastureBelowGroundOrgan.cs [hunk 2]
```diff
@@ -2,91 +2,74 @@
...
-        private INutrient SoilNitrogen;
+        /// <summary>Soil nutrient model where these roots are growing.</summary>
+        private INutrient nutrient;
 
         private double[] dulMM;
         private double[] ll15MM;
 
         /// <summary>The NO3 solute.</summary>
-        public ISolute NO3 = null;
+        private ISolute no3 = null;
```
Comment (hol430) [inline Models/AgPasture/PastureBelowGroundOrgan.cs:37]: Why can't we use a link for no3 and nh4?

File: Models/AgPasture/PastureBelowGroundOrgan.cs (hunk 1 above)
Comment (hol353) [reply Models/AgPasture/PastureBelowGroundOrgan.cs:17]: This is a work in progress. I am part way through refactoring this organ to remove things like this. There will be more commits to this PR.

File: Models/AgPasture/PastureBelowGroundOrgan.cs (hunk 2 above)
Comment (hol353) [reply Models/AgPasture/PastureBelowGroundOrgan.cs:37]: Can't use links because for multi-zone simulations, a root organ will be told which zone its roots are in.
